api/envoy/api/v2/auth/cert.proto

```diff
@@ -172,8 +172,24 @@ message TlsSessionTicketKeys {
   repeated core.DataSource keys = 1 [(validate.rules).repeated = {min_items: 1}];
 }
 
-// [#next-free-field: 9]
+// [#next-free-field: 10]
 message CertificateValidationContext {
+  // Peer certificate verification mode.
+  enum TrustChainVerification {
+    // Perform default certificate verification (e.g., against CA / verification lists)
+    VERIFY_TRUST_CHAIN = 0;
+
+    // Connections where the certificate fails verification will be permitted.
+    // For HTTP connections, the certificate details will be unconditionally placed in the
+    // *x-forwarded-untrusted-client-cert* HTTP header.
```

Comment:
Do we want to be highly prescriptive here on how we handle the certificate details? Some folks might be good with this just being a fingerprint or an indicator of the untrusted status.

Reply:
This was an old comment - we no longer hoist cert info into the 'untrusted' header (from review in  PR #8248). There is now a route matching TlsContextMatchOption instead of looking for specific headers.